prqlc/prql-compiler/src/lib.rs

```diff
@@ -71,6 +71,16 @@
 //!
 //! * `serde_yaml`: adapts the `Serialize` implementation for [`prqlc_ast::expr::ExprKind::Literal`]
 //!   to `serde_yaml`, which doesn't support the serialization of nested enums
+//!
+//! ## Large binary sizes
+//!
+//! For Linux users, the binary produced by this crate will probably be quite large (>200MB) by default.
+//! That is because it includes a lot of debuginfo symbols from our parser. You can remove them by
+//! adding the following to `Cargo.toml`:
+//! ```toml
+//! [profile.release.package.prql-compiler]
+//! strip = "debuginfo"
```

Comment:
Super, TIL, thanks!

Reply:
Should we raised this upstream in chumsky? (am happy to if you think it makes sense)

Reply:
Though is the 200MB number accurate? If I `cargo install prqlc` in a linux container (which links to a version of the code before this PR), then I get ~20MB. (still not small!)

```
root@ec8355540866:/# ls -al /usr/local/cargo/bin/prqlc
-rwxr-xr-x 1 root root 17658344 Nov  8 20:29 /usr/local/cargo/bin/prqlc
```

Reply:
Hmm, not sure actually. After `cargo build --release` I got >200MB. Maybe `cargo install` is doing something differently?

Reply:
What file specifically is that large? `prqlc`?

Reply:
Running `cargo build --release` on a cloned version of the repo at 0.10.0, in `docker run -it rust`:

```
root@ec8355540866:/prql# ls -al target/release/
total 35200
drwxr-xr-x   7 root root     4096 Nov  8 22:17 .
drwxr-xr-x   4 root root     4096 Nov  8 22:15 ..
-rw-r--r--   1 root root        0 Nov  8 22:15 .cargo-lock
drwxr-xr-x 170 root root    12288 Nov  8 22:16 .fingerprint
drwxr-xr-x  36 root root     4096 Nov  8 22:16 build
drwxr-xr-x   2 root root    36864 Nov  8 22:17 deps
drwxr-xr-x   2 root root     4096 Nov  8 22:15 examples
drwxr-xr-x   2 root root     4096 Nov  8 22:15 incremental
-rw-r--r--   1 root root     3498 Nov  8 22:15 libprql_compiler.d
-rw-r--r--   2 root root 18096752 Nov  8 22:15 libprql_compiler.rlib
-rwxr-xr-x   2 root root 17923488 Nov  8 22:17 prqlc
-rw-r--r--   1 root root     3603 Nov  8 22:17 prqlc.d
```

Reply:
My bad, it is the debug build:

```
➜  prql git:(main) ll target/release/prqlc*
-rwxr-xr-x 2 aljaz users  17M nov  9 07:52 target/release/prqlc
-rw-r--r-- 1 aljaz users 5,1K nov  9 07:52 target/release/prqlc.d
➜  prql git:(main) ll target/debug/prqlc*
-rwxr-xr-x 2 aljaz users 353M nov  9 07:54 target/debug/prqlc
-rw-r--r-- 1 aljaz users 5,1K nov  9 07:54 target/debug/prqlc.d
```

Reply:
After this PR:
```
➜  prql git:(main) ll target/release/prqlc*
-rwxr-xr-x 2 aljaz users 7,5M nov  9 07:56 target/release/prqlc
-rw-r--r-- 1 aljaz users 5,1K nov  9 07:52 target/release/prqlc.d
```

Reply:
OK cool!

Should I remove the docstring? I think we can keep the configs on `release` though (though I don't know the tradeoffs; do we lose tracebacks on panics etc...)

Reply:
Yes, doc string can be changed to 20MB.

We do lose core dumps (which could be useful if there is a segfault), but panics work as before, even when using the 7MB binary.

We also lose the ability to link against the executable, which is not something you normally do anyways (if it  is even possible)